webSites/webSites-war/src/main/java/org/silverpeas/components/websites/servlets/design/SiteDesignActionHandler.java

```diff
@@ -0,0 +1,648 @@
+
+    while (ok && i < tabContexte.size()) {
+      String contenuContexte = tabContexte.get(i);
+      if (i < tab.size()) {
+        String contenu = tab.get(i);
+        if (contenuContexte.equals(contenu)) {
+          array.add(contenu);
+
+        } else {
+          ok = false;
+        }
+        i++;
+      } else {
+        ok = false;
+      }
+    }
+    return array;
+  }
+
+  private String sortReste(List<String> tab, List<String> tabCommun) {
+    /* tab = [id | rep1 | rep2 | rep3] */
+    /* tabCommun = [id | rep1] */
+    /* res = rep2/rep3 */
+    String res = "";
+
+    int indice = tabCommun.size();
+
+    while (indice < tab.size()) {
+      String contenu = tab.get(indice);
+      res += contenu + "/";
+      indice++;
+    }
+
+    if (!res.equals("")) {
+      res = res.substring(0, res.length() - 1);
+    }
+
+    return res;
+  }
+
+  private String parseCodeSupprImage(WebSiteSessionController scc, String code,
+      HttpServletRequest request, SettingBundle settings, String currentPath) {
+    String avant;
+    String apres;
+    int index;
+    String finChemin;
+    String image =
+        "<IMG border=0 src=\"http://" + getMachine(request) + "/" + settings.getString("Context") +
+            "/" + scc.getComponentId() + "/";
+    int longueurImage = 19 +
+        ("http://" + getMachine(request) + "/" + settings.getString("Context") + "/" +
+            scc.getComponentId() + "/").length();
+    index = code.indexOf(image);
+    if (index == -1) {
+      return code;
+    } else {
+      avant = code.substring(0, index + 19);
+      finChemin = code.substring(index + longueurImage);
+
+      int indexGuillemet = finChemin.indexOf("\"");
+      String absolute = finChemin.substring(0, indexGuillemet);
+
+      apres = finChemin.substring(indexGuillemet);
+      int indexSlash = absolute.lastIndexOf("/");
+      String fichier = absolute.substring(indexSlash + 1);
+
+      String deb = absolute.substring(0, indexSlash);
+      List<String> tab = buildTab(deb + "/");
+
+      /* id/rep1 */
+      String cheminContexte = finNode(scc, currentPath);
+      List<String> tabContexte = buildTab(cheminContexte + "/");
+      List<String> tabCommun = sortCommun(tabContexte, tab);
+      String reste = sortReste(tab, tabCommun);
+      int nbPas = tabContexte.size() - tabCommun.size();
+      String relatif = "";
+      int i = 0;
+      while (i < nbPas) {
+        relatif += "../";
+        i++;
+      }
+
+      if (reste.equals("")) {
+        relatif += fichier;
+      } else {
+        relatif += reste + "/" + fichier;
+      }
+      apres = relatif + apres;
+      return (avant + parseCodeSupprImage(scc, apres, request, settings, currentPath));
+    }
+  }
+
+  private String parseCodeSupprHref(WebSiteSessionController scc, String code, String currentPath) {
+    String avant;
+    String apres;
+    int index;
+    String href = "<A href=\""; /* longueur de chaine = 9 */
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Move this trailing comment on the previous empty line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3ATrailingCommentCheck)